src/reconcile/rec_write.c

```diff
@@ -648,6 +649,15 @@ __rec_init(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags, WT_SALVAGE_COO
     r->update_modify_cbt.ref = ref;
     r->update_modify_cbt.iface.value_format = btree->value_format;
 
+    /*
+     * FIXME: cache the stable timestamp used to check if the durable timestamps in prepared updates
+     * can be discarded (until PM-1524 completes and durable timestamps are stored in data pages).
+     */
+    WT_ORDERED_READ(r->stable_ts, S2C(session)->txn_global.stable_timestamp);
```

Comment:
Is the ordered read necessary here? If so could you update the comment to describe why?

Reply:
@michaelcahill I have the same question. Can you explain it?

Reply:
The goal is to choose the largest timestamp we can that is less than or equal to the next checkpoint.  Once a checkpoint is running, it's clear that we should use the global `checkpoint_timestamp`.  When no checkpoint is running, we should use `stable_timestamp`.

The problem the ordered read is attempting to solve (and it isn't perfect) is when eviction races with a checkpoint copying `stable_timestamp` to `checkpoint_timestamp`.  The ordered read is necessary but not sufficient.  Without it, we could see that no checkpoint is running and then copy `stable_timestamp`.  But in the meantime a checkpoint could have started and chosen an earlier value.  So it makes the window smaller, but a race is still possible.

Since none of this code should survive PM-1524, it's probably not necessary to try too hard here.  I'm fine with having it just copy `stable_timestamp` without the ordered read.

Reply:
Fine - I suspected that was the case. If it's a best effort I'm good with the change. Let's get it merged and work on the correct solution in the prepare project.